server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderTests.java

```diff
@@ -521,10 +521,10 @@ public void testDiskThresholdWithAbsoluteSizes() {
         clusterState = startInitializingShardsAndReroute(strategy, clusterState);
 
         logShardStates(clusterState);
-        // primary shard already has been relocated away
-        assertThat(clusterState.getRoutingNodes().node(nodeWithPrimary).size(), equalTo(0));
-        // node with increased space still has its shard
-        assertThat(clusterState.getRoutingNodes().node(nodeWithoutPrimary).size(), equalTo(1));
+        assertThat(
+            clusterState.getRoutingNodes().node(nodeWithPrimary).size() + clusterState.getRoutingNodes().node(nodeWithoutPrimary).size(),
+            equalTo(1)
+        );
```

Comment:
This assertion assumed that primary (allocated on a node with a higher disk usage) is moved before replica (allocated on a node with a lower disk usage). In fact we do not have such guarantee and this is determined y iteration order.